### ISSUE TITLE
A11y: view, copy, etc controls for items are not keyboard focusable/operable

### ISSUE THREAD

**Issue by patrickhlauke** [bitwarden/browser#1986]
## Describe the Bug

Currently, the various controls for each entry (to view, copy username, copy password, copy verification code, etc) are only operable with the mouse. For keyboard users, it's only possible to set focus/activate the entry itself (to autofill), with no way to reach and trigger the separate controls.

This is due to the controls being coded as `<span>` elements, which don't receive keyboard focus.

```
<a appstopclick="" class="box-content-row box-content-row-flex ng-star-inserted" href="#" title="Auto-fill - github.com (private)">

...

<span appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="View" aria-label="View"> <i aria-hidden="true" class="fa fa-lg fa-list-alt"></i> </span>

<span appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="Copy username" aria-label="Copy username"> <i aria-hidden="true" class="fa fa-lg fa-user"></i> </span>

<span appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="Copy password" aria-label="Copy password"> <i aria-hidden="true" class="fa fa-lg fa-key"></i> </span>

<span appstopclick="" appstopprop="" class="row-btn disabled ng-star-inserted" title="Copy verification code" aria-label="Copy verification code"> <i aria-hidden="true" class="fa fa-lg fa-clock-o"></i> </span>

...

</a>
```

## Steps To Reproduce

1. Open the popup
2. Try to reach the various controls using the keyboard (e.g. using `Tab` to move focus to them)

## Expected Result

* being able to reach those controls and activate them using the keyboard

## Actual Result

* only the items themselves can be focused, but not the individual controls

## Screenshots or Videos

![An item with its controls](https://user-images.githubusercontent.com/895831/127736313-c12fab67-dc03-46d6-80f7-47a607651763.png)

## Environment

- Operating system: Windows 10
- Browser: Chrome 92
- Build Version (go to "Settings" → "About" in the app): 1.51.1

## Additional Context

This will require a bit of thoughts. Currently, these controls are inside an overarching `<a>` element (for each entry). Suggest that the approach here should be:

* split each entry up, so that the outer control (for autofill) only encompasses the name (or make a separate control for autofill)
* turn that autofill control into a `<button>` (rather than a link)
* turn the `<span>` elements into `<button>` elements
* ideally, associate the controls with the name of the entry, to give them content, using `aria-describedby`, or adding it explicitly in the `aria-label` that's already there
* if a control is disabled (as is the case in my example with the "Copy verification code"), just set the `<button>` to be `disabled`

Roughly, the ideal structure would be:

```
<button appstopclick="" class="box-content-row box-content-row-flex ng-star-inserted"  title="Auto-fill - github.com (private)">
  ...
</button>

<button appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="View" aria-label="View - github.com (private)"> ... </button>

<button appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="Copy username" aria-label="Copy username - github.com (private)"> ... </button>

<button appstopclick="" appstopprop="" class="row-btn ng-star-inserted" title="Copy password" aria-label="Copy password - github.com (private)"> ... </button>

<button disabled appstopclick="" appstopprop="" class="row-btn disabled ng-star-inserted" title="Copy verification code" aria-label="Copy verification code - github.com (private)"> ... </button>
```

**Comment by IvanVas** [comment]
I second this. It's a big deal for keyboard-only workflow.

**Comment by patrickhlauke** [comment]
realise though that once you have a large number of these cards, it'll make keyboard navigation more cumbersome. there are of course alternative approaches, like treating each card as a single tab stop "toolbar", and then having keyboard navigation to those buttons with a cursor (as long as they're correctly exposed when focused - roving `tabindex` or `aria-activedescendant` approaches come to mind). happy to help brainstorm some approaches if it helps...

**Comment by IvanVas** [comment]
> realise though that once you have a large number of these cards, it'll make keyboard navigation more cumbersome. there are of course alternative approaches, like treating each card as a single tab stop "toolbar", and then having keyboard navigation to those buttons with a cursor (as long as they're correctly exposed when focused - roving `tabindex` or `aria-activedescendant` approaches come to mind). happy to help brainstorm some approaches if it helps...

Arrows-based (2D) navigation is pretty good IMO. Not sure if it's viable from impl perspective though.

**Comment by patrickhlauke** [comment]
possibly a dual approach: make the card/entry itself a tab stop, and its set of controls a second tab stop implemented as an [aria practices 1.2 toolbar](https://www.w3.org/TR/wai-aria-practices-1.2/#toolbar)

**Comment by patrickhlauke** [comment]
hotkeys aren't really a great solution, as there can be problems with assistive technologies intercepting keypresses in certain situations/depending on how they're implemented. they then also immediately require keyboard users and assistive technology users having to first learn how they can reach controls/activate functionalities that are immediately reachable by mouse/touch users ... not the best experience. they can certainly be included, but i'd say only on top of a robust keyboard focus approach.
